src/main/java/org/team1540/robot2025/subsystems/intake/IntakeIOReal.java

```diff
@@ -0,0 +1,91 @@
+
+    private final MotionMagicVoltage motorRequest = new MotionMagicVoltage(0).withSlot(0);
+
+    // clockwise to intake, counter-clockwise to spit out
+    private final SparkMax neo = new SparkMax(IntakeConstants.NEO_ID, SparkLowLevel.MotorType.kBrushless);
+
+    public IntakeIOReal() {
+
+        TalonFXConfiguration spinTalonFXConfigs = new TalonFXConfiguration();
+        TalonFXConfiguration pivotTalonFXConfigs = new TalonFXConfiguration();
+
+        spinTalonFXConfigs.CurrentLimits.withStatorCurrentLimitEnable(true);
+        spinTalonFXConfigs.CurrentLimits.withStatorCurrentLimit(IntakeConstants.SPIN_FALCON_CURRENT_LIMIT);
+        spinTalonFXConfigs.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        spinTalonFXConfigs.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+
+        pivotTalonFXConfigs.CurrentLimits.withStatorCurrentLimitEnable(true);
+        pivotTalonFXConfigs.CurrentLimits.withStatorCurrentLimit(IntakeConstants.PIVOT_FALCON_CURRENT_LIMIT);
+        spinTalonFXConfigs.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+
+        Slot0Configs spinConfigs = pivotTalonFXConfigs.Slot0;
+        Slot1Configs pivotConfigs = pivotTalonFXConfigs.Slot1;
+
+        pivotConfigs.kS = IntakeConstants.INTAKE_KS;
+        pivotConfigs.kV = IntakeConstants.INTAKE_KV;
+        pivotConfigs.kA = IntakeConstants.INTAKE_KA;
+        pivotConfigs.kP = IntakeConstants.INTAKE_KP;
+        pivotConfigs.kI = IntakeConstants.INTAKE_KI;
+        pivotConfigs.kD = IntakeConstants.INTAKE_KD;
+
+        MotionMagicConfigs motionMagicConfigs = pivotTalonFXConfigs.MotionMagic;
+
+        motionMagicConfigs.MotionMagicCruiseVelocity = IntakeConstants.INTAKE_CRUISE_VELOCITY;
+        motionMagicConfigs.MotionMagicAcceleration = IntakeConstants.INTAKE_ACCELERATION;
+
+        spinFalcon.getConfigurator().apply(spinConfigs);
+        pivotFalcon.getConfigurator().apply(pivotConfigs);
+
+        pivotFalcon.setPosition(0);
+    }
+
+    @Override
+    public void setPivot(Rotation2d rotations) {
+        pivotFalcon.setControl(motorRequest.withPosition(rotations.getRotations()));
+    }
+
+    @Override
+    public void setRollerSpeed(double speed) {
+        spinFalcon.set(speed);
+    }
+
+    @Override
+    public void setNEOSpeed(double speed) {
+        neo.set(speed);
+    }
+
+    @Override
+    public void updateInputs(CoralIntakeInputs inputs) {
+
+        inputs.spinMotorPosition = spinFalcon.getPosition().getValueAsDouble();
+        inputs.spinMotorVelocityRPS = spinFalcon.getVelocity().getValueAsDouble();
+        inputs.spinMotorAppliedVolts = spinFalcon.getMotorVoltage().getValueAsDouble();
+
+        inputs.pivotMotorPosition = pivotFalcon.getPosition().getValueAsDouble();
+        inputs.pivotMotorVelocityRPS = pivotFalcon.getVelocity().getValueAsDouble();
+        inputs.pivotMotorAppliedVolts = pivotFalcon.getMotorVoltage().getValueAsDouble();
```

Comment:
Cache the status signals from the falcons and store them as fields in this class, and refresh them periodically in this method.